collector/factory.go

```diff
@@ -0,0 +1,57 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package collector // import "go.opentelemetry.io/ebpf-profiler/collector"
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/xconsumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/xreceiver"
+
+	"go.opentelemetry.io/ebpf-profiler/collector/internal"
+	"go.opentelemetry.io/ebpf-profiler/internal/controller"
+)
+
+var (
+	typeStr = component.MustNewType("otelreceiver")
+
+	errInvalidConfig = errors.New("invalid config")
+)
+
+// NewFactory creates a factory for the receiver.
+func NewFactory() receiver.Factory {
```

Comment:
What about renaming this function to `NewDefaultFactory()`?
```suggestion
// NewDefaultFactory creates a factory for the receiver.
func NewDefaultFactory() receiver.Factory {
```

Currently, users can not configure this factory and default values are used. Just thinking about the future, where there should be a configurable `receiver.Factory`, then it could be done (in the future) with `func NewFactory(cfg component.CreateDefaultConfigFunc, options ...FactoryOption ) receiver.Factory {..}`

Reply:
The method needs to be called `NewFactory()` with no option for OCB to work properly.